### DISCUSSION TITLE
Use Chatfred with LocalAI server?

### DISCUSSION THREAD

**Discussion by s3frank** [chrislemke/ChatFred#82] [Ideas]
I am keen to use ChatFred with a local model. I have been trying my luck a bit with that but running into some issues around the history that is being communicated to the server. I have tried setting it to zero but that didn't work.

Anyone else tried to use something like LocalAI with ChatFred?

https://github.com/go-skynet/LocalAI

**Comment by chrislemke** [comment]
Hey @s3frank!

very cool idea. Since local models getting more and more famous this could improve ChatFred even more. I guess quite a lot of work is needed. I will investigate more in this direction. 

Since you are already playing about it. Maybe you are interested in creating a PR?

**Comment by s3frank** [comment]
I will have a look, not sure yet due to other time commitments.
But given that you use OpenAI api and things like LocalAI expose that also
I think it may not be that much work to support it.

I will revert later after spending some more time on it first.

Great work you are doing mate, thank you!

Frank

On Sat, Jul 8, 2023 at 11:00 AM Chris Lemke ***@***.***>
wrote:

> Hey @s3frank <https://github.com/s3frank>!
>
> very cool idea. Since local models getting more and more famous this could
> improve ChatFred even more. I guess quite a lot of work is needed. I will
> investigate more in this direction.
>
> Since you are already playing about it. Maybe you are interested in
> creating a PR?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chrislemke/ChatFred/issues/75#issuecomment-1627023617>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQW7CRCQWCLGZLLWBHIXVLXPEVSRANCNFSM6AAAAAA2BVYDQI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Best regards,

-FF

**Comment by chrislemke** [comment]
> But given that you use OpenAI api and things like LocalAI expose that also
I think it may not be that much work to support it.

Good point. With that in mind, the implementation should be pretty straightforward.

**Comment by elsatch** [comment]
I think the only change needed would be to expose the OpenAI URL in the configuration, in the same way as the OpenAPI key. Then you could point it to any compatible OpenAI local server and... that's it!

**Comment by elsatch** [comment]
Reviewing the code and configuration, I've just discovered that there is a field called "Custom Open API URL". If you place your OpenAI compatible endpoint info here, it works. ( p.e. http://localhost:1234/v1 )

My main concern right now, is that you can't pass the custom model to the call. But, so far the server responses ok!
